galsim/interpolatedimage.py

```diff
@@ -550,6 +546,182 @@ def __setstate__(self, d):
+
+    def __init__(self, real_kimage, imag_kimage, k_interpolant=None, stepk=None,
+                 gsparams=None):
+
+        # make sure real_kimage, imag_kimage are really `Image`s, are floats, and are congruent.
+        if not isinstance(real_kimage, galsim.Image) or not isinstance(imag_kimage, galsim.Image):
+            raise ValueError("Supplied kimage is not an Image instance")
+        if ((real_kimage.dtype != np.float32 and real_kimage.dtype != np.float64)
+            or (imag_kimage.dtype != np.float32 and imag_kimage.dtype != np.float64)):
+            raise ValueError("Supplied image is not an image of floats or doubles!")
+        if real_kimage.bounds != imag_kimage.bounds:
+            raise ValueError("Real and Imag kimages must have same bounds.")
+        if real_kimage.scale != imag_kimage.scale:
+            raise ValueError("Real and Imag kimages must have same scale.")
+
+        # Make sure any `wcs`s are `PixelScale`s.
+        if ((real_kimage.wcs is not None
+             and not isinstance(real_kimage.wcs, galsim.PixelScale))
+            or (imag_kimage.wcs is not None
+                and not isinstance(imag_kimage.wcs, galsim.PixelScale))):
+            raise ValueError("Real and Imag kimage wcs's must be PixelScale's or None.")
+
+        # Check for Hermitian symmetry properties of real_kimage and imag_kimage
+        shape = real_kimage.array.shape
+        # If image is even-sized, ignore first row/column since in this case not every pixel has
+        # a symmetric partner to which to compare.
+        bd = galsim.BoundsI(real_kimage.xmin + (1 if shape[1]%2==0 else 0),
+                            real_kimage.xmax,
+                            real_kimage.ymin + (1 if shape[0]%2==0 else 0),
+                            real_kimage.ymax)
+        if not (np.allclose(real_kimage[bd].array,
+                            real_kimage[bd].array[::-1,::-1])
+                or np.allclose(imag_kimage[bd].array,
+                               -imag_kimage[bd].array[::-1,::-1])):
+            raise ValueError("Real and Imag kimages must form a Hermitian complex matrix.")
+
+        # Check edges of input Images to see if gsparams.maxk_threshold criterion is met.  We
+        # can't actually know the values outside of the provide Images, of course, so this is
+        # only a warning.
+        if gsparams is None:
+            maxk_threshold = galsim.GSParams().maxk_threshold
+        else:
+            maxk_threshold = gsparams.maxk_threshold
+        edge = np.zeros_like(real_kimage.array, dtype=bool)
+        edge[[0,-1],:] = True # top and bottom
+        edge[:,[0,-1]] = True # left and right
+        edge_max = np.max(np.abs(real_kimage.array[edge] + 1j*imag_kimage.array[edge]))
+        if (edge_max / real_kimage.array.max()) > maxk_threshold:
+            import warnings
+            warnings.warn(
+                "Provided Real and Imag kimages do not meet GSParams.maxk_threshold criterion.")
```

Comment:
>I'm thinking this is pointing to a bug in drawKImage.

Hmm...  Looks like it missed by one pixel:
```
a = galsim.Gaussian(flux=1.7, sigma=2.3)
re, im = a.drawKImage()
print re.array.shape # (14, 14)
print re.array[7,7] # 1.7
print re.array[7, 7+6] # 0.0100987
print a.kValue(0, 6*a.stepK()) # 0.0100987
print a.kValue(0, 6*a.stepK())/1.7 # 0.005940
print a.kValue(0, 7*a.stepK())/1.7 # 0.000933
```

I think it makes sense to truncate images such that they include the nearest pixel beyond where the threshold gets crossed, rather than at the last pixel inside of the threshold.  Perhaps increment by 2 rows/columns in SBProfile::getGoodImageSize()?